src/gc/objecthandle.cpp

```diff
@@ -792,9 +797,18 @@ bool Ref_InitializeHandleTableBucket(HandleTableBucket* bucket, void* context)
         // No free slot.
         // Let's create a new node
         NewHolder<HandleTableMap> newMap;
-        newMap = new HandleTableMap;
+        newMap = new (nothrow) HandleTableMap;
+        if (!newMap)
+        {
+            return false;
+        }
+
+        newMap->pBuckets = new (nothrow) HandleTableBucket * [ INITIAL_HANDLE_TABLE_ARRAY_SIZE ];
+        if (!newMap->pBuckets)
+        {
+            return false;
```

Comment:
I thought this is what `HandleTableBucketHolder` does [here](https://github.com/dotnet/coreclr/blob/master/src/gc/objecthandle.cpp#L609)? The only control flow I see that suppresses the release of that holder is the one that returns true.